### ISSUE TITLE
[ISSUE] Issue importing`databricks_metastore_data_access` resource

### ISSUE THREAD

**Issue by andrewmetcalfbede** [databricks/terraform-provider-databricks#2749]
After failing to update the owner of one of our metastore storage credentials (https://github.com/databricks/terraform-provider-databricks/issues/2748). I decided to update the owner manually and re-import the credential into Terraform.  The import is successful despite yet Terraform wants to destroy and re-create the resource when running a subsequent plan.

The Terraform configuration matches the parameters of the storage credential.

### Configuration
```hcl
resource "databricks_metastore_data_access" "metastoreDataAccess" {
  count        = var.resourceCount
  metastore_id = var.metastoreId
  name         = var.metastoreName
  is_default   = var.isDefault
  owner        = var.owner
  azure_managed_identity {
    access_connector_id = var.accessConnectorId
  }
}
```

```
 make 'import "module.mod_unityCatalogMetastore.module.metastoreDataAccess.databricks_metastore_data_access.metastoreDataAccess[0]" "obfuscated-metastore-id|obfuscated-name"'
```

### Expected Behavior
Terraform reports no changes once the resource has been imported

### Actual Behavior
Terraform wants to recreate the resource

```hcl
  # module.mod_unityCatalogMetastore.module.metastoreDataAccess.databricks_metastore_data_access.metastoreDataAccess[0] must be replaced
-/+ resource "databricks_metastore_data_access" "metastoreDataAccess" {
      ~ id           = "obfuscated-metastore-id|obfuscated-name" -> (known after apply)
        name         = "obfuscated-name"
      + owner        = "obfuscated-owner"
        # (2 unchanged attributes hidden)

      + azure_managed_identity { # forces replacement
          + access_connector_id = "/subscriptions/obfuscated-sub/resourceGroups/obfuscated-rg/providers/Microsoft.Databricks/accessConnectors/obfuscated-name" # forces replacement
          + credential_id       = (known after apply)
        }

      + databricks_gcp_service_account {
          + credential_id = (known after apply)
          + email         = (known after apply)
        }
    }

Plan: 1 to add, 0 to change, 1 to destroy.

 ➜  ~ curl -s --header "Authorization: Bearer ${DATABRICKS_TOKEN}" --request GET "${DATABRICKS_HOST}api/2.0/accounts/${DATABRICKS_ACCOUNT_ID}/metastores/${METASTORE_ID}/storage-credentials" | jq .
{
  "storage_credentials": [
    {
      "name": "obfuscated-name",
      "azure_managed_identity": {
        "access_connector_id": "/subscriptions/obfuscated-sub/resourceGroups/obfuscated-rg/providers/Microsoft.Databricks/accessConnectors/obfuscated-name",
        "credential_id": "dobfuscated-id"
      },
      "read_only": false,
      "owner": "obfuscated-owner",
      "id": "obfuscated-cred-id",
      "metastore_id": "obfuscated-metastore-id",
      "created_at": 1695721883473,
      "created_by": "obfuscated-sp",
      "updated_at": 1695995426644,
      "updated_by": "obfuscated-sp",
      "used_for_managed_storage": true,
      "full_name": "obfuscated-name",
      "securable_type": "STORAGE_CREDENTIAL",
      "securable_kind": "STORAGE_CREDENTIAL_AZURE_MI"
    }
  ]
}
```

### Steps to Reproduce
<!-- Please list the steps required to reproduce the issue, for example:
1. Add value to "owner" parameter on `databricks_metastore_data_access` resource
2. Remove storage credential from Terraform state
3. Update "owner" value of credential via API 
4. Import storage credential into Terraform
5. Run a Terraform plan

### Terraform and provider versions
```
make version
/Users/met/Documents/github/Bede.Terraform/bin/1.0.11/terraform -v
Terraform v1.0.11
on darwin_amd64
+ provider registry.terraform.io/databricks/databricks v1.27.0
+ provider registry.terraform.io/hashicorp/azuread v2.42.0
+ provider registry.terraform.io/hashicorp/azurerm v3.73.0
```

### Debug Output
Not provided

### Important Factoids
No

**Comment by nkvuong** [comment]
Same root cause as #2697 (the read is not returning the attributes)

**Comment by mgyucht** [comment]
This is fixed by #2804. Thanks @nkvuong. @andrewmetcalfbede can you confirm this is no longer an issue on the latest provider version?

**Comment by andrewmetcalfbede** [comment]
Thanks @mgyucht unfortunately I can't test this as I no longer have the issue after recreating the underlying resources, but thanks for sorting it.
